lib/jsonwp-proxy/proxy.js

```diff
@@ -299,25 +299,28 @@ class JWProxy {
   }
 
   async proxyReqRes (req, res) {
-    let [response, body] = await this.proxyCommand(req.originalUrl, req.method, req.body);
+    const [response, resBodyObj] = await this.proxyCommand(req.originalUrl, req.method, req.body);
 
     res.headers = response.headers;
     res.set('content-type', response.headers['content-type']);
     // if the proxied response contains a sessionId that the downstream
     // driver has generated, we don't want to return that to the client.
     // Instead, return the id from the request or from current session
-    body = util.safeJsonParse(body);
-    if (body && body.sessionId) {
+    let isResBodyModified = false;
+    if (_.has(resBodyObj, 'sessionId')) {
```

Comment:
could there ever be a situation where `resBodyObj` is a JSON string rather than an object here?

Reply:
isPlainObject check should take care of this

Reply:
as for now proxyCommand will always return an object as the second argument